### PR TITLE
Add force download

### PR DESCRIPTION
**Pull request by laka3000**

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1 @@
+quit
```
Comment (sobrinho) [inline .byebug_history]: Remove this file.

File: README.md [hunk 2]
```diff
@@ -426,6 +426,7 @@ There's also a helper for generating image tags:
 <%= attachment_image_tag(@user, :profile_image, :fill, 300, 300) %>
 ```
 
+
```
Comment (sobrinho) [inline README.md]: Remove this empty line.

File: README.md [hunk 3]
```diff
@@ -434,6 +435,12 @@ no file has been uploaded:
 <%= attachment_image_tag(@user, :profile_image, :fill, 300, 300, fallback: "default.png") %>
 ```
 
+You can also set the URL to force the download of the uploaded file:
+
+``` erb
+<%= link_to "Download", attachment_url(@user, :profile_image, :force_download => true) %>
```
Comment (sobrinho) [inline README.md]: Use `force_download: true` instead of `:force_download => true`.

File: lib/refile.rb [hunk 4]
```diff
@@ -320,6 +322,10 @@ def file_url(file, *args, expires_at: nil, host: nil, prefix: nil, filename:, fo
         base_path += "?expires_at=#{expires_at.to_i}" # UNIX timestamp
       end
 
+      if (!!force_download)
```
Comment (sobrinho) [inline lib/refile.rb]: Use `if force_download` instead of `if (!!force_download)`.

File: lib/refile.rb [hunk 5]
```diff
@@ -382,16 +388,17 @@ def presign_url(backend, host: nil, prefix: nil)
     # @param [String, nil] format          A file extension to be appended to the URL
     # @param [String, nil] host            Override the host
     # @param [String, nil] prefix          Adds a prefix to the URL if the application is not mounted at root
-    # @return [String, nil]                The generated URL
-    def attachment_url(object, name, *args, expires_at: nil, host: nil, prefix: nil, filename: nil, format: nil)
+    # @param [String, nil] expires_at      Adds a sulfix to the URL that sets the expiration time of the URL
+    # @param [String, nil] force_download  Adds a sulfix to the URL to force the download of the file when URL is acce    # @return [String, nil]                The generated URL
```
Comment (sobrinho) [inline lib/refile.rb]: The `# @return ...` have moved to the force download line.

File: lib/refile/app.rb [hunk 6]
```diff
@@ -177,5 +178,10 @@ def not_expired?(params)
       params["expires_at"].nil? ||
         (Time.at(params["expires_at"].to_i) > Time.now)
     end
+
+    def force_download?(params)
+      !params["force_download"].nil?
+    end
+
```
Comment (sobrinho) [inline lib/refile/app.rb]: Avoid this empty line at the end of the class.

File: lib/refile/app.rb [hunk 7]
```diff
@@ -136,8 +136,9 @@ def stream_file(file)
       end
 
       filename = Rack::Utils.unescape(request.path.split("/").last)
+      disposition = force_download?(params) ? 'attachment' : 'inline'
```
Comment (sobrinho) [inline lib/refile/app.rb]: Use double-quote instead of single-quote to keep consistency.

**Comment by laka3000** [comment]
Fixed code style. Added test to force_download argument.

**Comment by laka3000** [comment]
Addressed issues raised by rubocop

**Comment by sobrinho** [comment]
Thank you very much, merged!

**Comment by jorgevbo** [comment]
Could you upload a release version to RubyGems? Thanks!

**Comment by sobrinho** [comment]
I'm planning a new release in the next month, we have a few issues to close before releasing the official one.

Feel free to use the master version until then.

**Pull request merged**